tool/tctl/common/tctl.go

```diff
@@ -94,12 +94,21 @@ func Run(commands []CLICommand, loadConfigExt LoadConfigFn) {
 		commands[i].Initialize(app, cfg)
 	}
 
-	// these global flags apply to all commands
 	var ccf GlobalCLIFlags
+
+	// Initially, set the config file path to the default.
+	// If this is overridden by environment variable, update the path.
+	ccf.ConfigFile = defaults.ConfigFilePath
+	configFileEnvar, isSet := os.LookupEnv(defaults.ConfigFileEnvar)
+	if isSet {
+		ccf.ConfigFile = configFileEnvar
+	}
+
+	// these global flags apply to all commands
 	app.Flag("debug", "Enable verbose logging to stderr").
 		Short('d').
 		BoolVar(&ccf.Debug)
-	app.Flag("config", fmt.Sprintf("Path to a configuration file [%v]", defaults.ConfigFilePath)).
+	app.Flag("config", fmt.Sprintf("Path to a configuration file [%v]. Can also be set via the %v environment variable.", defaults.ConfigFilePath, defaults.ConfigFileEnvar)).
```

Comment:
Yeah - I'm not using `Envar` directly here because of the fact that `ExistingFileVar` is the way `/etc/teleport.yaml` is currently loaded into `ccf.ConfigFile` with `tctl` in the default case. I couldn't see a way to stack both `Envar` and `ExistingFileVar` to give a precedence order, so it seemed simpler to just handle it manually ahead of time.

Reply:
Not sure I follow. `ccf.ConfigFile` is only set to default (`defaults.ConfigFilePath`) if `--config` is not used. Also, it is not enforced on kingpin level but is taken into account when applying configuration (if `ccf.ConfigFile` happens to be empty).
With `Envar`, the attribute will be set to whatever the value of the environment variable was unless `--config` is also specified in which case, its value is used instead (since Envar works as kind of a default). If none of these are used, the default value is used.

Reply:
I tried this:

```go
app.Flag("config", fmt.Sprintf("Path to a configuration file [%v]. Can also be set via the %v environment variable.", defaults.ConfigFilePath, defaults.ConfigFileEnvar)).
    Short('c').
    Envar(defaults.ConfigFileEnvar).
    StringVar(&ccf.ConfigFile)
```

It seems the issue is that `Envar` doesn't overwrite the string if the environment variable is blank, i.e. if you use `TELEPORT_CONFIG_FILE="" tctl...` then it continues to read `/etc/teleport.yaml`. That's the behaviour that I need to suppress somehow.

If I remove the code above that which sets `ccf.ConfigFile = defaults.ConfigFilePath` then `/etc/teleport.yaml` is not read at all when there are local credentials available in the `~/.tsh` directory, which is a change from the current behaviour. This may be more desirable as one could explicitly pass `-c /etc/teleport.yaml` if that file needs to be read, but as it's an alteration from `tctl`'s current behaviour for a fairly uncommon use case, I'm not keen to introduce the change.

Reply:
Ok then, does not look like it's worth it.